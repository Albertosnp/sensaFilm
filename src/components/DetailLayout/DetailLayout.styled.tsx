import styled from '@emotion/styled';
import breakpoints from '@/assets/styles/breakpoints';
import { rem } from '@/assets/styles/_mixins';

const DetailLayoutStyled = styled.section<{ backgroudImg: string }>`
  position: absolute;
  top: 0;

  height: 60%;
  width: 100%;
  background-position: center top;
  background-repeat: no-repeat;
  background-size: cover;
  background-image: ${(props) => props.backgroudImg};

  .DetailLayout {
    &__info {
      padding: 10%;
      position: relative;
      z-index: 1;
      top: 150px;
      color: #fff;

      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      gap: 50px;

      &__imgPoster {
        width: 250px;
        height: auto;
        box-shadow: rgba(0, 0, 0, 0.644) 3px 3px 6px;
        border-radius: 5px;
      }

      &__detailsBlock {
        display: flex;
        flex-direction: column;
        align-self: flex-end;
        gap: 10px;

        &__title {
          font-size: ${rem(31)};
          padding-bottom: 20px;
        }

        &__overview {
          text-shadow: black 0.1em 0.1em 0.2em;
          max-width: 600px;
          font-size: ${rem(14)};
        }
      }
    }

    &__gradient {
      position: absolute;
      top: 0;
      left: 0;
      height: 100%;
      width: 100%;
      background: linear-gradient(
        to bottom,
        transparent 0%,
        transparent 60%,
        #232228 100%
      );
    }

    &__detailTitle {
      font-size: ${rem(18)};
      padding-bottom: 10px;
    }
  }
  @media (min-width: ${rem(breakpoints.tablet)}) {
  }
  @media (min-width: ${rem(breakpoints.tablet_wide)}) {
    height: 70%;
    .DetailLayout {
      &__info {
        flex-direction: row;
      }
    }
  }
  @media (min-width: ${rem(breakpoints.desktop)}) {
  }
`;
export default DetailLayoutStyled;
