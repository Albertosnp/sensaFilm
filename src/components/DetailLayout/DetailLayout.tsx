import { BASE_URL } from '@/config';
import { ReactNode } from 'react';
import DetailLayoutStyled from './DetailLayout.styled';

interface DetailLayoutProps {
  backdrop_path: string;
  poster_path: string;
  overview: string;
  title: string;
  children: ReactNode;
}

export const DetailLayout = ({
  backdrop_path,
  poster_path,
  overview,
  title,
  children,
}: DetailLayoutProps) => {
  const style = {
    backgroundImage: `url('${BASE_URL}${backdrop_path}')`,
  };
  return (
    <DetailLayoutStyled
      className="DetailLayout"
      backgroudImg={style.backgroundImage}
    >
      <div className="DetailLayout__gradient" />
      <article className="DetailLayout__info">
        <img
          src={`${BASE_URL}${poster_path}`}
          className="DetailLayout__info__imgPoster"
        />

        <div className="DetailLayout__info__detailsBlock">
          <h3 className="DetailLayout__info__detailsBlock__title">{title}</h3>
          <h6 className="DetailLayout__detailTitle">Sinopsis:</h6>
          <p className="DetailLayout__info__detailsBlock__overview">
            {overview}
          </p>
        </div>
      </article>
      {children}
    </DetailLayoutStyled>
  );
};
