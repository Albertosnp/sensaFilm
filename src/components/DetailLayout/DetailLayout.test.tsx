import React from "react";
import { createRoot } from "react-dom/client";
import { render } from "@testing-library/react";
import { vi } from "vitest";
import { DetailLayout } from "./DetailLayout";

describe("DetailLayout", () => {
  const defaultProps = {
    backdrop_path: "backdrop_path",
    poster_path: "poster_path",
    overview: "overview",
    title: "title",
    children: <div>children</div>,
  };
  const getComponent = (props = defaultProps) => <DetailLayout {...props} />;
  const setup = (props = defaultProps) => render(getComponent(props));

  it("Should render without crashing", () => {
    const div = document.createElement("div");
    createRoot(div).render(getComponent());
    createRoot(div).unmount();
  });

  it("Should match snapshot", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
