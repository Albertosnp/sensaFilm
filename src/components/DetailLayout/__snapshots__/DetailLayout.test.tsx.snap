// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DetailLayout > Should match snapshot 1`] = `
<div>
  <section
    class="DetailLayout css-s5ahiv"
  >
    <div
      class="DetailLayout__gradient"
    />
    <article
      class="DetailLayout__info"
    >
      <img
        class="DetailLayout__info__imgPoster"
        src="https://image.tmdb.org/t/p/w500/poster_path"
      />
      <div
        class="DetailLayout__info__detailsBlock"
      >
        <h3
          class="DetailLayout__info__detailsBlock__title"
        >
          title
        </h3>
        <h6
          class="DetailLayout__detailTitle"
        >
          Sinopsis:
        </h6>
        <p
          class="DetailLayout__info__detailsBlock__overview"
        >
          overview
        </p>
      </div>
    </article>
    <div>
      children
    </div>
  </section>
</div>
`;
